### PR TITLE
Made username and color in AutoMod mod messages use correct values

### PR DESCRIPTION
**Pull request by zneix**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

- Switched to category logging in PubsubClient.cpp
- Introduced displayName and color in ActionUser struct
- Renamed name to login in ActionUser to make it a bit more clear and avoid confusion with displayName
- Cleaned up some code in couple places

Final product:

if you see this i von

### CODE REVIEW

File: src/messages/MessageBuilder.cpp [hunk 1]
```diff
@@ -117,19 +117,20 @@ std::pair<MessagePtr, MessagePtr> makeAutomodMessage(
...
     builder
-        .emplace<TextElement>(
-            action.target.name + ":", MessageElementFlag::BoldUsername,
-            MessageColor(QColor("red")), FontStyle::ChatMediumBold)
-        ->setLink({Link::UserInfo, action.target.name});
+        .emplace<TextElement>(action.target.displayName + ":",
+                              MessageElementFlag::BoldUsername,
+                              MessageColor(QColor(action.target.color)),
+                              FontStyle::ChatMediumBold)
+        ->setLink({Link::UserInfo, action.target.displayName});
```
Comment (pajlada) [inline src/messages/MessageBuilder.cpp:128]: The link should point at the login name

File: src/messages/MessageBuilder.cpp [hunk 2]
```diff
@@ -343,16 +345,16 @@ MessageBuilder::MessageBuilder(const UnbanAction &action)
...
 
-    this->message().timeoutUser = action.target.name;
+    this->message().timeoutUser = action.target.login;
 
     QString text;
 
-    this->emplaceSystemTextAndUpdate(action.source.name, text)
-        ->setLink({Link::UserInfo, action.source.name});
+    this->emplaceSystemTextAndUpdate(action.target.login, text)
+        ->setLink({Link::UserInfo, action.target.login});
```
Comment (pajlada) [inline src/messages/MessageBuilder.cpp:353]: You've replaced source with target here - is this intentional?

File: src/messages/MessageBuilder.cpp [hunk 3]
```diff
@@ -369,35 +371,35 @@ MessageBuilder::MessageBuilder(const AutomodUserAction &action)
     {
         case AutomodUserAction::AddPermitted: {
             text = QString("%1 added %2 as a permitted term on AutoMod.")
-                       .arg(action.source.name)
+                       .arg(action.target.login)
```
Comment (pajlada) [inline src/messages/MessageBuilder.cpp:374]: You've replaced source with target here - is this intentional?

File: src/messages/MessageBuilder.cpp [hunk 4]
```diff
@@ -369,35 +371,35 @@ MessageBuilder::MessageBuilder(const AutomodUserAction &action)
...
                        .arg(action.message);
         }
         break;
 
         case AutomodUserAction::AddBlocked: {
             text = QString("%1 added %2 as a blocked term on AutoMod.")
-                       .arg(action.source.name)
+                       .arg(action.target.login)
```
Comment (pajlada) [inline src/messages/MessageBuilder.cpp:381]: You've replaced source with target here, is that intentional?

File: src/messages/MessageBuilder.cpp [hunk 5]
```diff
@@ -369,35 +371,35 @@ MessageBuilder::MessageBuilder(const AutomodUserAction &action)
...
                        .arg(action.message);
         }
         break;
 
         case AutomodUserAction::RemovePermitted: {
             text = QString("%1 removed %2 as a permitted term on AutoMod.")
-                       .arg(action.source.name)
+                       .arg(action.target.login)
```
Comment (pajlada) [inline src/messages/MessageBuilder.cpp:388]: You've replaced source with target here, is that intentional?

File: src/messages/MessageBuilder.cpp [hunk 6]
```diff
@@ -369,35 +371,35 @@ MessageBuilder::MessageBuilder(const AutomodUserAction &action)
...
                        .arg(action.message);
         }
         break;
 
         case AutomodUserAction::RemoveBlocked: {
             text = QString("%1 removed %2 as a blocked term on AutoMod.")
-                       .arg(action.source.name)
+                       .arg(action.target.login)
```
Comment (pajlada) [inline src/messages/MessageBuilder.cpp:395]: You've replaced source with target here, is that intentional?

File: src/providers/twitch/PubsubClient.cpp [hunk 7]
```diff
@@ -1428,23 +1431,45 @@ void PubSub::handleMessageResponse(const rapidjson::Value &outerData)
...
                     return;
                 }
-                action.target = ActionUser{sender_id, sender_login};
-                qDebug() << action.msgID;
+                QString senderDisplayName;
+                if (!rj::getSafe(senderData, "display_name", senderDisplayName))
+                {
+                    qCDebug(chatterinoPubsub)
+                        << "Failed to get sender display name";
+                    return;
```
Comment (pajlada) [inline src/providers/twitch/PubsubClient.cpp:1454]: Could we just handle this gracefully by setting displayName to login?
EDIT: I see we do comparisons with login below, so we would need to take that into consideration too

File: src/providers/twitch/PubsubClient.cpp [hunk 8]
```diff
@@ -1428,23 +1431,45 @@ void PubSub::handleMessageResponse(const rapidjson::Value &outerData)
...
+                }
+                QString senderColor;
+                if (!rj::getSafe(senderData, "chat_color", senderColor))
+                {
+                    qCDebug(chatterinoPubsub)
+                        << "Failed to get sender color, might be a grey-name";
+                    // return; // color might be empty if user is a grey-name, don't freak out
```
Comment (pajlada) [inline src/providers/twitch/PubsubClient.cpp:1461]: Let's make sure we handle this case gracefully too - either leave the color empty and ensure we handle it, or set it to a default color. No need to leave a debug message here

**Review by pajlada — CHANGES_REQUESTED** [review]
Some small car thoughts 🚙

File: src/providers/twitch/PubsubClient.cpp [hunk 9]
```diff
@@ -1428,23 +1431,45 @@ void PubSub::handleMessageResponse(const rapidjson::Value &outerData)
...
+                }
+                // handling non-ascii usernames
+                if (QString::compare(senderLogin, senderDisplayName,
+                                     Qt::CaseInsensitive) != 0)
+                {
+                    senderDisplayName =
+                        QString("%1(%2)").arg(senderLogin, senderDisplayName);
```
Comment (Mm2PL) [inline src/providers/twitch/PubsubClient.cpp:1468]: This should respect the username style setting

**Review by Mm2PL — CHANGES_REQUESTED** [review]

File: src/messages/MessageBuilder.cpp (hunk 1 above)
Comment (zneix) [reply src/messages/MessageBuilder.cpp:128]: Good catch, thank you!

File: src/messages/MessageBuilder.cpp (hunk 2 above)
Comment (zneix) [reply src/messages/MessageBuilder.cpp:353]: Here, and in all other cases with the same comment I must've used bulk-replace in a wrong way, thanks for noticing!

File: src/messages/MessageBuilder.cpp [hunk 10]
```diff
@@ -268,13 +269,13 @@ MessageBuilder::MessageBuilder(const BanAction &action, uint32_t count)
...
-        if (!action.source.name.isEmpty())
+        if (!action.target.login.isEmpty())
         {
             this->emplaceSystemTextAndUpdate("by", text);
             this->emplaceSystemTextAndUpdate(
-                    action.source.name + (action.reason.isEmpty() ? "." : ":"),
+                    action.target.login + (action.reason.isEmpty() ? "." : ":"),
                     text)
-                ->setLink({Link::UserInfo, action.source.name});
+                ->setLink({Link::UserInfo, action.target.login});
```
Comment (pajlada) [inline src/messages/MessageBuilder.cpp:278]: This replacement has changed `source` to `target`

File: src/messages/MessageBuilder.cpp [hunk 11]
```diff
@@ -287,29 +288,30 @@ MessageBuilder::MessageBuilder(const BanAction &action, uint32_t count)
     {
         if (action.isBan())
         {
-            this->emplaceSystemTextAndUpdate(action.source.name, text)
-                ->setLink({Link::UserInfo, action.source.name});
+            this->emplaceSystemTextAndUpdate(action.target.login, text)
+                ->setLink({Link::UserInfo, action.target.login});
```
Comment (pajlada) [inline src/messages/MessageBuilder.cpp:292]: This replacement has changed `source` to `target`

File: src/messages/MessageBuilder.cpp [hunk 12]
```diff
@@ -287,29 +288,30 @@ MessageBuilder::MessageBuilder(const BanAction &action, uint32_t count)
...
                 this->emplaceSystemTextAndUpdate(
                     QString("\"%1\".").arg(action.reason), text);
             }
         }
         else
         {
-            this->emplaceSystemTextAndUpdate(action.source.name, text)
-                ->setLink({Link::UserInfo, action.source.name});
+            this->emplaceSystemTextAndUpdate(action.target.login, text)
+                ->setLink({Link::UserInfo, action.target.login});
```
Comment (pajlada) [inline src/messages/MessageBuilder.cpp:311]: This replacement has changed `source` to `target`

File: src/messages/MessageBuilder.cpp [hunk 13]
```diff
@@ -369,35 +371,35 @@ MessageBuilder::MessageBuilder(const AutomodUserAction &action)
...
-                       .arg(action.source.name)
+                       .arg(action.source.login)
                        .arg(action.message);
         }
         break;
 
         case AutomodUserAction::Properties: {
             text = QString("%1 modified the AutoMod properties.")
-                       .arg(action.source.name);
+                       .arg(action.target.login);
```
Comment (pajlada) [inline src/messages/MessageBuilder.cpp:402]: This replacement has changed `source` to `target`

**Review by pajlada — CHANGES_REQUESTED** [review]

File: src/providers/twitch/PubsubClient.cpp (hunk 8 above)
Comment (zneix) [reply src/providers/twitch/PubsubClient.cpp:1460]: Waiting for #2974 to be merged in - then I'll handle this gracefully by setting a random color based on the `senderId`, exactly how we handle greynames in Twitch channels (while also respecting "Colorize greynames" setting).

File: src/providers/twitch/PubsubClient.cpp (hunk 9 above)
Comment (zneix) [reply src/providers/twitch/PubsubClient.cpp]: I believe I sorted this out in bb004cc
proof

File: src/providers/twitch/PubsubClient.cpp (hunk 8 above)
Comment (zneix) [reply src/providers/twitch/PubsubClient.cpp]: Should be resolved in 8e1a70eb; I tested and it respects the setting correctly, even while changing it on the go.

**Review by pajlada — APPROVED** [review]
:+1:

**Pull request merged**